schneiderWallbox/integrationpluginschneiderwallbox.cpp

```diff
@@ -0,0 +1,546 @@
+        qCDebug(dcSchneiderElectric()) << "Reachable state changed" << reachable;
+        if (reachable) {
+            schneiderWallboxTcpConnection->initialize();
+        } else {
+            thing->setStateValue("connected", false);
+        }
+    });
+
+    // Only during setup
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::initializationFinished, info, [this, thing, schneiderWallbox, monitor, info](bool success){
+        if (success) {
+            qCDebug(dcSchneiderElectric()) << "Schneider wallbox initialized.";
+            m_schneiderDevices.insert(thing, schneiderWallbox);
+            m_monitors.insert(thing, monitor);
+            info->finish(Thing::ThingErrorNoError);
+        } else {
+            hardwareManager()->networkDeviceDiscovery()->unregisterMonitor(monitor);
+            schneiderWallbox->deleteLater();
+            info->finish(Thing::ThingErrorHardwareFailure, QT_TR_NOOP("Could not initialize the communication with the wallbox."));
+        }
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::updateFinished, thing, [schneiderWallboxTcpConnection, thing](){
+        qCDebug(dcSchneiderElectric()) << "Update finished:" << thing->name() << schneiderWallboxTcpConnection;
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::initializationFinished, thing, [thing](bool success){
+        if (success) {
+            thing->setStateValue("connected", true);
+        }
+    });
+
+    // Handle property changed signals
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::chargeTimeChanged, this, [thing](quint32 chargeTime){
+        quint32 chargeTimeMinutes{chargeTime / 60};
+        qCDebug(dcSchneiderElectric()) << thing << "charging time changed" << chargeTimeMinutes << "minutes";
+        thing->setStateValue("chargeTime", chargeTimeMinutes);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::cpwStateChanged, this, [thing, this](SchneiderWallboxModbusTcpConnection::CPWState cpwState){
+        qCDebug(dcSchneiderElectric()) << thing << "CPW state changed" << cpwState;
+        setCpwState(thing, cpwState);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::lastChargeStatusChanged, this, [thing, this](SchneiderWallboxModbusTcpConnection::LastChargeStatus lastChargeStatus){
+        qCDebug(dcSchneiderElectric()) << thing << "Last charge status changed" << lastChargeStatus;
+        setLastChargeStatus(thing, lastChargeStatus);
+    });
+
+    connect(schneiderWallboxTcpConnection, &SchneiderWallboxModbusTcpConnection::stationEnergyChanged, this, [thing](quint32 stationEnergyInWattHours){
+        double stationEnergyInKilowattHours{(static_cast<double>(stationEnergyInWattHours)) / 1000.0};
```

Comment:
I think unless using a *very* old gcc (which wouldn't support the list initializer), the static_cast to double is not needed and the `.0` in the `/1000.0` will do the right thing.